servers/rendering/rendering_device.cpp

```diff
@@ -432,7 +432,7 @@ void RenderingDevice::_staging_buffer_execute_required_action(StagingBuffers &p_
 Error RenderingDevice::buffer_copy(RID p_src_buffer, RID p_dst_buffer, uint32_t p_src_offset, uint32_t p_dst_offset, uint32_t p_size) {
 	ERR_RENDER_THREAD_GUARD_V(ERR_UNAVAILABLE);
 
-	ERR_FAIL_COND_V_MSG(draw_list, ERR_INVALID_PARAMETER,
+	ERR_FAIL_COND_V_MSG(draw_list && draw_list->viewport_set == true, ERR_INVALID_PARAMETER,
```

Comment:
Along with my other suggestion, I'd merely change draw list to have an active boolean and just check for it. Same with compute list. We should not need to dynamically allocate them at all.